File: src/main/java/net/hectus/neobb/turn/legacy_game/warp/LTCliffWarp.java
package net.hectus.neobb.turn.legacy_game.warp;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.Clazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.WaterClazz;
import org.bukkit.World;

import java.util.List;

public class LTCliffWarp extends LWarpTurn {
    public LTCliffWarp(World world) { super(world, "legacy-cliff"); }
    public LTCliffWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "legacy-cliff"); }

    @Override
    public void apply() {
        super.apply();
        new Buff.Luck(Buff.BuffTarget.YOU, 15);
    }

    @Override
    public int chance() {
        return 67;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, ColdClazz.class, WaterClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/legacy_game/warp/LTCliffWarp.kt
package net.hectus.neobb.modes.turn.legacy_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.matrix.structure.PlacedStructure
import net.hectus.neobb.matrix.structure.StaticStructure
import net.hectus.neobb.matrix.structure.StaticStructures
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.Clazz
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.WaterClazz
import net.hectus.neobb.player.NeoPlayer
import kotlin.reflect.KClass

class LTCliffWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : LWarpTurn(data, cord, "cliff", player) {
    override val staticStructure: StaticStructure = StaticStructures.Legacy.Warp.CLIFF

    override val cost: Int = 4
    override val chance: Double = 66.66
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, ColdClazz::class, WaterClazz::class)
    override val temperature: Temperature = Temperature.NORMAL

    override fun apply() {
        Luck(15).apply(player!!)
    }
}
